src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,842 @@
+                return (AveragingMode)(GetRegisterBits(Register.Mode, BitMask.ModeAVG));
+            }
+
+            set
+            {
+                SetRegisterBits(Register.Mode, BitMask.ModeAVG, (uint)value);
+            }
+        }
+
+        /// <summary>
+        /// Set the filter output data rate select bits. The 10 bits of data programmed into these bits determine the filter cutoff frequency, the position of the first notch of the filter, and the output data rate for the part.
+        /// </summary>
+        public ushort Filter
+        {
+            get
+            {
+                return (ushort)(GetRegisterBits(Register.Mode, BitMask.ModeFS));
+            }
+
+            set
+            {
+                SetRegisterBits(Register.Mode, BitMask.ModeFS, value);
+            }
+        }
+
+        /// <summary>
+        /// The value of the offset calibration register
+        /// </summary>
+        public uint Offset
+        {
+            get
+            {
+                return ReadRegisterValue(Register.Offset) & (uint)BitMask.Offset;
+            }
+
+            set
+            {
+                SetRegisterBits(Register.Offset, BitMask.Offset, value);
+            }
+        }
+
+        /// <summary>
+        /// The value of the full-scale calibration register
+        /// </summary>
+        public uint FullScale
+        {
+            get
+            {
+                return ReadRegisterValue(Register.FullScale) & (uint)BitMask.FullScale;
+            }
+
+            set
+            {
+                SetRegisterBits(Register.FullScale, BitMask.FullScale, value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the metadata class for the device
+        /// </summary>
+        /// <returns></returns>
+        public static IDeviceMetadata GetDeviceMetadata()
+        {
+            return Ad7193.metadata;
+        }
+
+        /// <summary>
+        /// Initializes the ADC
+        /// </summary>
+        /// <param name="spiDevice">The SPI device to initialize the ADC on</param>
+        public Ad7193(SpiDevice spiDevice)
+        {
+            if ((spiDevice.ConnectionSettings.Mode & metadata.ValidSpiModes) != spiDevice.ConnectionSettings.Mode)
+            {
+                throw new Exception("SPI device must be in SPI mode 3 in order to work with AD7193.");
+            }
+
+            if (spiDevice.ConnectionSettings.ClockFrequency > metadata.MaximumSpiFrequency)
+            {
+                throw new Exception($"SPI device must have a lower clock frequency, because AD7193's maximum operating SPI frequncy is {metadata.MaximumSpiFrequency} Hz.");
+            }
+
+            _spiDevice = spiDevice;
+
+            Reset();
+        }
+
+        /// <summary>
+        /// Resets the ADC
+        /// </summary>
+        public void Reset()
+        {
+            for (int i = 0; i < 6; i++)
+            {
+                _spiDevice.Write(new byte[] { 0xFF });
```

Comment:
2 requests:
- can this be done by writing 6 bytes in 1 write instead
- please change to use Span. I.e.:

```csharp
public void Reset()
{
    Span<byte> bytes = stackalloc byte[6];
    bytes.Fill(0xFF); // or just use it as an array, i.e. bytes[0] = 0xFF;
    // ...
}
```

this way you don't allocate 6 arrays on this call